app/models/requests/request_decorator.rb

```diff
@@ -87,13 +87,22 @@ def only_aeon?
 
     def location_label
       return "" if holding.blank?
-      label = holding["library"]
-      label += " - #{holding['location']}" if holding["location"].present?
+      if any_items? && requestable.first.item["in_temp_library"]
+        label = holding["current_library"]
+        label += " - #{holding['current_location']}" if holding["current_location"].present?
+      else
+        label = holding["library"]
+        label += " - #{holding['location']}" if holding["location"].present?
+      end
       label
     end
 
     def holding
-      holdings[mfhd]
+      if any_items? && requestable.first.item["in_temp_library"]
```

Comment:
Just to keep it consistent with other checks, what about:
```suggestion
      if any_items? && requestable.first.item.temp_loc?
```

Reply:
thank you! I updated another one I missed !